src/otx/algo/callbacks/adaptive_train_scheduling.py

```diff
@@ -78,3 +83,36 @@ def on_train_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
     def _get_adaptive_interval(self, iter_per_epoch: int, max_interval: int) -> int:
         """Get adaptive interval."""
         return max(round(math.exp(self.decay * iter_per_epoch) * max_interval), 1)
+
+    def _change_lr_scheduler_frequency(self, lr_configs: list[LRSchedulerConfig], adaptive_interval: int) -> None:
+        """Change the frequency of LRscheduler.
+
+        Since adaptive interval changes the validation interval, the frequency of LRscheduler also
+        should be changed according to the adaptive interval.
+        """
+        for config in lr_configs:
+            if hasattr(config, "frequency"):
+                msg = (
+                    "The frequency of LRscheduler will be changed due to the effect of adaptive interval: "
+                    f"{config.frequency} --> {adaptive_interval}."
+                )
+                log.warning(msg)
+                config.frequency = adaptive_interval
```

Comment:
```suggestion
         self.revert_frequency = []
         ...
                saved_frequency = config.frequency
                config.frequency = adaptive_interval

                def _revert():
                    config.frequency = saved_frequency
                self.revert_frequency += [_revert]

...

    def on_train_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
        for revert in self.revert_frequency:
            revert()
```

Reply:
To maintain the same behavior as the validation interval, please add the revert action at `on_train_end()`

Reply:
Done, https://github.com/openvinotoolkit/training_extensions/pull/2857/commits/7803e5ac6b1e0e8b3199f34ab5155b998bd76bdb